gson/src/main/java/com/google/gson/internal/Streams.java

```diff
@@ -39,13 +40,22 @@ private Streams() {
 
   /**
    * Takes a reader in any state and returns the next value as a JsonElement.
+   *
+   * @param requireEndDocument whether there must not be any trailing data after
+   *    the JsonElement
    */
-  public static JsonElement parse(JsonReader reader) throws JsonParseException {
```

Comment:
Right, before #1959 `RuntimeTypeAdapterFactory` was using the internal `Streams` class. I have now added back that method, but marked it as deprecated to make users aware that it is declared in an internal Gson class, I hope that is ok.

> Then of course we could continue to call it from `RuntimeTypeAdapterFactory` instead of the new `(..., false)` version.

The `RuntimeTypeAdapterFactory` variant which was still using this method was actually a copy of the class from the `extras` module. I have now removed that copy (and the enclosing test class `RuntimeTypeAdapterFactoryFunctionalTest`) and have moved the adjusted test code to the `RuntimeTypeAdapterFactoryTest` class in the `extras` module to avoid code duplication.